base_layer/mmr/src/error.rs

```diff
@@ -37,4 +38,5 @@ pub enum MerkleMountainRangeError {
     HashNotFound(usize),
     // A request was out of range
     OutOfRange,
+    MerkleProofError(MerkleProofError),
```

Comment:
The initial idea was that the dependency hierarchy is MerkleProof > MMR. i.e. The MMR struct doesn't know anything about MerkleProofs, those are built on top of MMR.

Reply:
By exposing the `MerkleMountainRange` of the `MutableMmr`, I was able to move this `MerkleProofError` to `ChainStorageError`. Now the `MerkleProof::for_leaf_node` function is called from `fetch_mmr_proof` in `MemoryDatabase`.